ipa/ipa_ssh.py

```diff
@@ -0,0 +1,184 @@
+    """Class for testing instances in Azure."""
+
+    def __init__(self,
+                 access_key_id=None,  # Not used in SSH
+                 account_name=None,  # Not used in SSH
+                 cleanup=None,
+                 config=None,
+                 description=None,
+                 distro_name=None,
+                 early_exit=None,
+                 history_log=None,
+                 image_id=None,  # Not used in SSH
+                 inject=None,
+                 instance_type=None,  # Not used in SSH
+                 ip_address=None,
+                 log_level=30,
+                 no_default_test_dirs=False,
+                 provider_config=None,
+                 region=None,  # Not used in SSH
+                 results_dir=None,
+                 running_instance_id=None,  # Not used in SSH
+                 secret_access_key=None,  # Not used in SSH
+                 service_account_file=None,  # Not used in SSH
+                 ssh_key_name=None,  # Not used in SSH
+                 ssh_private_key=None,
+                 ssh_user=None,
+                 subnet_id=None,  # Not used in SSH
+                 test_dirs=None,
+                 test_files=None,
+                 timeout=None,
+                 vnet_name=None,  # Not used in SSH
+                 vnet_resource_group=None  # Not used in SSH
+                 ):
+        """Initialize Azure Provider class."""
+        super(SSHProvider, self).__init__('ssh',
+                                          cleanup,
+                                          config,
+                                          description,
+                                          distro_name,
+                                          early_exit,
+                                          history_log,
+                                          image_id,
+                                          inject,
+                                          instance_type,
+                                          log_level,
+                                          no_default_test_dirs,
+                                          provider_config,
+                                          region,
+                                          results_dir,
+                                          running_instance_id,
+                                          test_dirs,
+                                          test_files,
+                                          timeout)
+
+        # Cannot cleanup SSH instance
+        self.cleanup = False
+
+        self.ssh_private_key = (
+            ssh_private_key or
+            self._get_value(ssh_private_key, config_key='ssh_private_key')
+        )
+        if not self.ssh_private_key:
+            raise SSHProviderException(
+                'SSH private key file is required to connect to instance.'
+            )
+        else:
+            self.ssh_private_key = os.path.expanduser(
+                self.ssh_private_key
+            )
```

Comment:
Hmm there is some inconsistency with this implementation vs. my assumption based on the doc. Pleas reconcile.

Reply:
As per previous comments you cannot pass in a private key string. The CLI validates the string is a path and the file exists.

Reply:
```
@click.option(
    '--ssh-private-key',
    type=click.Path(exists=True),
    help='SSH private key file for accessing instance.'
)
```